### ISSUE TITLE
Latest commit (1.3.12-23-gd8fb007) usage show « GLib-WARNING » ...

### ISSUE THREAD

**Issue by Creteil** [bkw777/mainline#273]
```
bigbob@bigbob-ux331un:~$ sudo mainline --install-latest
mainline 1.3.21
Distribution: Ubuntu 22.04.2 LTS
Architecture: amd64
Running kernel: 6.4.0-060400-generic
Updating from: 'https://kernel.ubuntu.com/~kernel-ppa/mainline/'
Fetching individual kernel indexes...

(process:11960): GLib-WARNING **: 22:03:42.018: GChildWatchSource: Exit status of a child process was requested but ECHILD was received by waitpid(). See the documentation of g_child_watch_source_new() for possible causes.
Found installed: linux-image-unsigned-6.4.0-060400-generic (running)  
No updates
mainline: done
bigbob@bigbob-ux331un:~$
```

**Comment by bkw777** [comment]
Does this happen every time or randomly?
Can you reprproduce with -v 3?
Don't worry if you can't, I think I need to rig up some kind of torture test for exec_async() to find a race condition somewhere.

**Comment by bkw777** [comment]
I think I figured it out, and it's something I added recently. Trying to be extra robust actually was wrong, I added an extra way to detect the end of the process because I felt the original way of "when reading stdout fails" was not 100% certain, and now I understand a note in the manual.

I added this after spawn_async_with_pipes() :
```
ChildWatch.add(child_pid, (pid, status) => {
	Process.close_pid(pid);
	is_running = false;
});
```
and the manual for ChildWatch says 
> If you obtain pid from spawn_async or spawn_async_with_pipes you will need to pass g_spawn_do_not_reap_child as flag to the spawn function for the child watching to work.

and I indeed do not have that flag on the spawn_async_with_pipes() call.

So I should either add that flag, or remove the pid watch.
